client/src/pages/Calendar.js

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from "react";
+import Alert from "react-bootstrap/Alert";
+import "./Calendar.css";
+
+const Calendar = () => {
+	const [bookingsData, setBookingsData] = useState([]);
+	const [error, setError] = useState(null);
+	const [isLoading, setIsLoading] = useState(false);
+
+	useEffect(() => {
+		setIsLoading(true);
+		fetch("/api/bookings?filter=2weeks")
+			.then((response) => {
+				if (!response.ok) {
+					throw Error(
+						"There was an error getting the calendar data. Please try again."
+					);
+				}
+				return response.json();
+			})
+			.then((data) => {
+				setBookingsData(data);
+			})
+			.catch((err) => {
+				setError(err.message);
+			})
+			.finally(() => {
+				setIsLoading(false);
+			});
+	}, []);
+
+	if (isLoading) {
+		return (
+			<div className="d-flex justify-content-center">
+				<div className="spinner-grow">
+					<span className="visually-hidden">Loading...</span>
+				</div>
+			</div>
+		);
+	}
+	if (error != null) {
+		return <Alert variant="danger">{error}</Alert>;
+	}
+// Each element has the structure { date: Date, clients: [] }
+	const pickUpDates = [];
+
+const isSameDate = (dateA, dateB) => {
```

Comment:
This function doesn't depend on any React state, so it doesn't need to be in the React component. We can move this out of the component to line 4 since we don't need to create this function on every component render

Reply:
Thanks Coung.